crates/ra_ide/src/completion/complete_qualified_path.rs

```diff
@@ -91,8 +93,21 @@ pub(super) fn complete_qualified_path(acc: &mut Completions, ctx: &CompletionCon
                 }
             }
         }
+        Some(res @ PathResolution::TypeParam(_)) | Some(res @ PathResolution::SelfType(_)) => {
+            if let Some(typens) = res.in_type_ns() {
+                associated_items(ctx.db, typens, |_, _, assoc_id| {
```

Comment:
API-wise, maybe we should just put a wrapper for `associated_items` on `PathResolution`. Then we don't need a public `in_type_ns` method. Also the wrapper can turn the `TypeAliasId`s into `TypeAlias`es, which is the proper HIR facade wrapper. Finally, I think we don't even need to match here -- we can just unconditionally call that method on `res`, and it'll return items when it's the right kind of resolution.